src/SprykerEco/Yves/Braintree/Form/DataProvider/CheckoutShipmentFormDataProvider.php

```diff
@@ -105,14 +106,16 @@ protected function createAvailableShipmentChoiceList(QuoteTransfer $quoteTransfe
         $shipmentMethods = [];
 
         $shipmentMethodsTransfer = $this->getAvailableShipmentMethods($quoteTransfer);
-        foreach ($shipmentMethodsTransfer->getMethods() as $shipmentMethodTransfer) {
-            if (!isset($shipmentMethods[$shipmentMethodTransfer->getCarrierName()])) {
-                $shipmentMethods[$shipmentMethodTransfer->getCarrierName()] = [];
+        foreach ($shipmentMethodsTransfer->getShipmentMethods() as $shipmentMethodsTransfer) {
+            foreach ($shipmentMethodsTransfer->getMethods() as $shipmentMethodTransfer) {
+                if (!isset($shipmentMethods[$shipmentMethodTransfer->getCarrierName()])) {
```

Comment:
You are using `$shipmentMethodTransfer->getCarrierName()` 3x
You could store this in local var and use that 3x then.

Reply:
> You are using `$shipmentMethodTransfer->getCarrierName()` 3x
> You could store this in local var and use that 3x then.

that's not an improvement. https://boyet.com/blog/local-variables-are-evil-or-just-irritating/

Reply:
It actually is an improvement, we do this.

Reply:
That article was 10 years ago.
The code also becomes more readable and maintainable, which is usually a plus:
```php
//$shipmentMethods[$shipmentMethodTransfer->getCarrierName()][$description] = ...
$shipmentMethods[$carrierName][$description] = ...
```

Reply:
OK i did that)